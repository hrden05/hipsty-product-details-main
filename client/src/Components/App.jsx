import React from 'react';
import axios from 'axios';
import HeaderBar from './HeaderBar.jsx';
import ProductPreDescription from './ProductPreDescription.jsx';
import ProductDescription from './ProductDescription.jsx';
import ShippingAndReturns from './ShippingAndReturns.jsx';
import SellerData from './SellerData.jsx';
import { storeWave, preFooterWave, subscribeWave } from './svgFiles.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      store: [],
      product: [],
      loading: true,
    };
  }

  componentDidMount() {
    const randomStore = Math.floor(Math.random() * 100 + 1);
    axios.get('api/joshstore', {
      params: {
        randomStore,
      },
    })
      .then((response) => {
        // const randomStore = Math.floor(Math.random() * res.data.stores.length);
        this.setState({
          store: response.data.stores,
          product: response.data.product,
          loading: false,
        });

      })
      .catch((err) => {
        console.log(err.message)
    });
  };

  render() {
    if (this.state.loading) {
      return <h1>Loading... </h1>
    }
    return (
      <div className='body'>
        {/* <div className='headerBar'>Header Section</div>
        <div className='productPictures'>Product Pictures</div>
        <div className='productPurchase'>Product Purchase</div>
        <div className='productReviews'>Product Reviews</div> */}
        <div id='productDescription' className='productDescription'>
          <div>
            <ProductPreDescription store={this.state.store} product={this.state.product} />
          </div>
          <div>
            <ProductDescription store={this.state.store} product={this.state.product} />
          </div>
          <div>
            <ShippingAndReturns store={this.state.store} product={this.state.product} />
          </div>
          <div><SellerData store={this.state.store} /></div>
        </div>
        {/* <span className='storeInfoRoof'>{storeWave}</span>
        <div className='storeInfo'>storeInfo</div>
        <div className='adsSection'>ads</div>
        <div className='relatedProducts'>Related Products</div>
        <span className='subscribeRoof'>{subscribeWave}</span>
        <div className='subscribe'>Subscribe</div>
        <span className='preFooterRoof'>{preFooterWave}</span>
        <div className='preFooter'>Pre-footer area</div>
        <div className='footerBar'>Footer Section</div> */}
      </div>
    );
  }
}

export default App;
