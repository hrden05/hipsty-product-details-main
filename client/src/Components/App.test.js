import React from 'react';
import renderer from 'react-test-renderer';
import App from './App.jsx';

test('App changes ');
