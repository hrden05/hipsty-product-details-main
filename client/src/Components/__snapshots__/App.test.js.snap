// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App renders, with props passed in 1`] = `ShallowWrapper {}`;

exports[`App App renders, without any data 1`] = `ShallowWrapper {}`;

exports[`App App renders, without any props passing in 1`] = `ShallowWrapper {}`;

exports[`App App should update store state 1`] = `undefined`;
